api/v1alpha4/types.go

```diff
@@ -106,11 +106,27 @@ type VnetSpec struct {
 	// +optional
 	CIDRBlocks []string `json:"cidrBlocks,omitempty"`
 
+	// Peerings defines a list of peerings of the newly created virtual network with existing virtual networks.
+	// +optional
+	Peerings VnetPeerings `json:"peerings,omitempty"`
+
 	// Tags is a collection of tags describing the resource.
 	// +optional
 	Tags Tags `json:"tags,omitempty"`
 }
 
+// VnetPeeringSpec specifies an existing remote virtual network to peer with the newly created virtual network.
+type VnetPeeringSpec struct {
+	// ResourceGroup is the resource group name of the remote virtual network.
+	ResourceGroup string `json:"resourceGroup"`
```

Comment:
This looks to be required. Would there be value in defaulting this to the cluster resource group if not specified?